applications: nrf_desktop: Add a note about disabling BLE legacy pairing (#20619)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

    This change results from the Bluetooth subsystem transition to the PSA cryptographic API.
    The GATT database hash calculation now requires larger stack size.

>   * Support for Bluetooth LE legacy pairing is no longer enabled by default, because it's not secure.

Reviewer: ```suggestion
  * Support for Bluetooth LE legacy pairing is no longer enabled by default because it is not secure.
```

    This change results from the Bluetooth subsystem transition to the PSA cryptographic API.
    The GATT database hash calculation now requires larger stack size.

  * Support for Bluetooth LE legacy pairing is no longer enabled by default, because it's not secure.

>     Using Bluetooth LE legacy pairing introduces risk of eavesdropping.

Author: Can we add it as an example (this one is relevant for BLE HID use-case I think):

`Using Bluetooth LE legacy pairing introduces, among others, risk of eavesdropping.`

>     Supporting Bluetooth LE legacy pairing makes devices vulnerable for a downgrade attack.

Author: I tried to separate the `using` from `supporting` here

>     The :kconfig:option:`CONFIG_BT_SMP_SC_PAIR_ONLY` is enabled by default in Zephyr.
>     If you still need to support the Bluetooth LE legacy pairing you need to disable the option in configuration.

Reviewer: ```suggestion
    The :kconfig:option:`CONFIG_BT_SMP_SC_PAIR_ONLY` Kconfig option is enabled by default in Zephyr.
    If you still need to support the Bluetooth LE legacy pairing, you need to disable the option in the configuration.
```

>   * Support for Bluetooth LE legacy pairing is no longer enabled by default, because it's not secure.
>     Using Bluetooth LE legacy pairing introduces risk of eavesdropping.
>     Supporting Bluetooth LE legacy pairing makes device vulnerable for downgrade attack.
>     The :kconfig:option:`CONFIG_BT_SMP_SC_PAIR_ONLY` is enabled by default in Zephyr.
>     If you still need to support the Bluetooth LE legacy pairing you need to disable the option in configuration.

Reviewer 2: ```suggestion
  * Support for Bluetooth LE legacy pairing is no longer enabled by default, because it is not secure.
    Using Bluetooth LE legacy pairing introduces a risk of eavesdropping.
    Supporting Bluetooth LE legacy pairing makes devices vulnerable for downgrade attack.
    The :kconfig:option:`CONFIG_BT_SMP_SC_PAIR_ONLY` Kconfig option is enabled by default in Zephyr.
    If you still need to support the Bluetooth LE legacy pairing, you need to disable the option in the configuration.
```

Reviewer: Deleted my comments as its already covered in Pekka's suggestion.
